### DISCUSSION TITLE
how to load a wheel file in xonsh?

### DISCUSSION THREAD

**Discussion by m172021** [xonsh/xonsh#5485] [Q&A] [answered]
Using `maturin build`, I can build a Rust pyo3 crate into a wheel file: target/wheels/b_python_0.1.00-cp311-cpe11-manylinux.whl

I can load this wheel file fine from Python and call b_python.sum_as_string(2, 3)

How do I import this wheel file from xonsh so I can use it from within xonsh ?

Thanks!

EDIT: Complicating issues, I'm on nixos

**Comment by Spitfire1900** [comment, marked as answer]
Is `xpip install target/wheels/b_python_0.1.00-cp311-cpe11-manylinux.whl` suitable?

**Reply by m172021** [reply]
I could not figure out how to run xpip, but the following worked for me:

unzip whl (its just a zip file)
add the directory to sys.path (in xonsh)
import b_python

**Reply by anki-code** [reply]
`xpip` is a xonsh builtin to install packages in the env where you run xonsh. See also [how to install xonsh](https://github.com/anki-code/xonsh-cheatsheet/blob/main/README.md#install-xonsh).
